inc/cloudfront_media_purge/namespace.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Altis\Cloud\Cloudfront_Media_Purge;
+
+use Exception;
+use function Altis\get_aws_sdk;
+
+function bootstrap() {
+	add_action( 'delete_attachment', __NAMESPACE__ . '\\purge_media_file_cache', 100 );
+}
+
+/**
+ * Return an AWS CloudFront Client instance
+ *
+ * @return \Aws\CloudFront\CloudFrontClient
+ */
+function get_aws_client() {
+	return get_aws_sdk()->createCloudFront( [
+		'version' => 'latest',
+	] );
+}
+
+/**
+ * Purge an attachment file cache from CloudFront
+ *
+ * It sends a request to invalidate all sizes given for a media file
+ *
+ * @param int $media_id Attachment ID.
+ *
+ * @return bool
+ */
+function purge_media_file_cache( $media_id ) {
+	$meta = wp_get_attachment_metadata( $media_id );
+	if ( ! $meta ) {
+		return false;
+	}
+	$baseurl = '/tachyon/';
+	$basedir = trailingslashit( dirname( $meta['file'] ) );
+	$items   = [ $baseurl . $meta['file'] . '*' ];
+
+	if ( ! empty( $meta['sizes'] ) ) {
+		foreach ( $meta['sizes'] as $size => $size_meta ) {
+			$items[] = $baseurl . $basedir . $size_meta['file'] . '*';
+		}
+	}
```

Comment:
This won't quite work - if you want to purge the different sizes you'd need to use the `tachyon_url()` function. If the `*` is a wildcard I'd suggest just adding the base URL of the file and just stripping the file extension so all the suffixes like query params and `file-200x200.jpg` are removed in one go.